syntax = "proto3";

package user;

message User {
	int32 id = 1;
	bool role = 2;
	string name = 3;
	string email = 4;
	string address = 5;
	string password = 6;
	string phone_number = 7;
}

message UserCreateReq {
	bool role = 1;
	string name = 2;
	string email = 3;
	string address = 4;
	string password = 5;
	string phone_number = 6;
}
message UserCreateRes {
	string msg = 1;
	User user = 2;
}

message UserAllReq {
    
}
message UserAllRes {
	string msg = 1;
	repeated User user = 2;
}

message UserOneReq {
	int32 id = 1;
}
message UserOneRes {
	User user = 2;
}

message UserUpdateReq {
	int32 id = 1;
	bool role = 2;
	string name = 3;
	string email = 4;
	string address = 5;
	string password = 6;
	string phone_number = 7;
}
message UserUpdateRes {
	string msg = 1;
	User user = 2;
}

message UserDeleteReq {
	int32 id = 1;
}
message UserDeleteRes {
	string msg = 1;
}


service UserService {
	rpc create (UserCreateReq) returns (UserCreateRes);
	rpc readAll (UserAllReq) returns (UserAllRes);
	rpc readOne (UserOneReq) returns (UserOneRes);
	rpc update (UserUpdateReq) returns (UserUpdateRes);
	rpc delete (UserDeleteReq) returns (UserDeleteRes);
}